flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/source/reader/KafkaPartitionSplitReaderTest.java

```diff
@@ -248,6 +253,100 @@ public void testAssignEmptySplit() throws Exception {
+                createReader(
+                        props,
+                        InternalSourceReaderMetricGroup.mock(metricListener.getMetricGroup()));
+        // Add a committed offset split
+        reader.handleSplitsChanges(
+                new SplitsAddition<>(
+                        Collections.singletonList(
+                                new KafkaPartitionSplit(
+                                        new TopicPartition(TOPIC1, 0),
+                                        KafkaPartitionSplit.COMMITTED_OFFSET))));
+        // pendingRecords should have not been registered because of lazily registration
+        assertFalse(metricListener.getGauge(MetricNames.PENDING_RECORDS).isPresent());
+        // Trigger first fetch
+        reader.fetch();
+        final Optional<Gauge<Long>> pendingRecords =
+                metricListener.getGauge(MetricNames.PENDING_RECORDS);
+        assertTrue(pendingRecords.isPresent());
+        // Validate pendingRecords
+        assertNotNull(pendingRecords);
+        assertEquals(NUM_RECORDS_PER_PARTITION - 1, (long) pendingRecords.get().getValue());
+        for (int i = 1; i < NUM_RECORDS_PER_PARTITION; i++) {
+            reader.fetch();
+            assertEquals(NUM_RECORDS_PER_PARTITION - i - 1, (long) pendingRecords.get().getValue());
+        }
+    }
+
+    @Test
+    public void testUsingCommittedOffsetsWithLatestOffsetResetStrategy() throws Throwable {
+        final Properties props = new Properties();
+        props.setProperty(
+                ConsumerConfig.AUTO_OFFSET_RESET_CONFIG,
+                OffsetResetStrategy.LATEST.name().toLowerCase());
+        props.setProperty(
+                ConsumerConfig.GROUP_ID_CONFIG, "using-committed-offset-with-latest-offset-reset");
+        KafkaPartitionSplitReader reader =
+                createReader(props, UnregisteredMetricsGroup.createSourceReaderMetricGroup());
+        // Add empty latest offset reset split
+        final KafkaPartitionSplit latestOffsetResetEmptySplit =
+                new KafkaPartitionSplit(
+                        new TopicPartition(TOPIC1, 0),
+                        KafkaPartitionSplit.COMMITTED_OFFSET,
+                        KafkaPartitionSplit.LATEST_OFFSET);
+        final KafkaPartitionSplit latestOffsetResetNormalSplit =
+                new KafkaPartitionSplit(
+                        new TopicPartition(TOPIC2, 0), KafkaPartitionSplit.COMMITTED_OFFSET);
+
+        reader.handleSplitsChanges(
+                new SplitsAddition<>(
+                        Arrays.asList(latestOffsetResetEmptySplit, latestOffsetResetNormalSplit)));
+
+        // Fetch and check latest offset reset split is added to finished splits
+        RecordsWithSplitIds<ConsumerRecord<byte[], byte[]>> recordsWithSplitIds = reader.fetch();
+        assertTrue(
+                recordsWithSplitIds
+                        .finishedSplits()
+                        .contains(latestOffsetResetEmptySplit.splitId()));
```

Comment:
I doubt that this case is testing the expected behavior. If the reader doesn't act as excepted, for example the fetcher doesn't respect the `auto.offset.reset` config and starts from the earliest offset, the first split (latestOffsetResetEmptySplit) could also reach the end offset in `reader.fetch()`, be added to finished splits, and pass this case. Also I can't see any purpose of the second split (latestOffsetResetNormalSplit) here.

Similar to the case above, I think a correct way is to check if the position before any `fetch()` is at the latest offset. This is the actual expected behavior of the reader.

Reply:
`latestOffsetResetEmptySplit` will be added to the Finished split, but if there is no unfinished split `consumer.poll()` in the `reader.fetch()`  will throw an exception.
I also didn’t find out how to get the finished split without using `reader.fetch()` , so the test is done like this

Reply:
One possible solution: 
- Add a package-visible helper function to expose the consumer in `KafkaPartitionSplitReader`:
```java
@VisibleForTesting
KafkaConsumer<byte[], byte[]> consumer() {
    return consumer;
}
```
- Use `consumer.position()` in the test case to check the consuming offset before any fetch(), which is the actual starting offset of the reader.

Reply:
For the finished split, the reader will unassign the partition, and for the Kafka consumer, it is not allowed to get the offset of the unassign partition.

Reply:
What about making the split not bounded (not setting the ending offset)?

Reply:
Thank you for your code review, I will fix soon.